arrow2_convert/src/deserialize.rs

```diff
@@ -7,36 +7,36 @@ use crate::field::*;
 
 #[doc(hidden)]
 /// Type whose reference can be used to create an iterator.
-pub trait IterRef {
+pub trait RefIntoIterator: Sized {
     /// Iterator type.
-    type Iter<'a>: Iterator
+    type Iterator<'a>: Iterator
```

Comment:
The reason this was `Iter` is because this is the common naming e.g. in the standard library: `std::option::Iter`, `std::result::Iter`, etc, always a `*Iter` if it's a type (whereas it's `*Iterator` if it's a trait, like `DoubleEndedIterator`).

Definitely +1 on `RefIntoIterator`, but I think `Iter<'_>` is more consistent.

Reply:
> Do you think it is something we can potentially tackle as well in the GAT rework?

Yes, absolutely. I'll rework this PR based on your feedback, and either include this change in this PR or next one.